app/classifier/tasks/drawing/markings-renderer.cjsx

```diff
@@ -35,50 +35,89 @@ module.exports = React.createClass
-                containerRect: @props.containerRect
-                scale: @props.scale
-                disabled: isPriorAnnotation
-                selected: mark is @state.selection and not isPriorAnnotation
-                getEventOffset: @props.getEventOffset
-                preferences: @props.preferences
-
-              toolProps =
-                classification: @props.classification
-                mark: mark
-                details: toolDescription.details
-                color: toolDescription.color
-                size: toolDescription.size
-
-              toolMethods =
-                onChange: @handleChange.bind this, i
-                onSelect: @handleSelect.bind this, annotation, mark
-                onDeselect: @handleDeselect
-                onDestroy: @handleDestroy.bind this, annotation, mark
-
-              ToolComponent = drawingTools[toolDescription.type]
-              <ToolComponent key={mark._key} {...toolProps} {...toolEnv} {...toolMethods} />}
-          </g>}
+          if @props.workflow.configuration.multi_image_tool
+            <g key={annotation._key} className="marks-for-annotation" data-disabled={isPriorAnnotation || null}>
+                {for mark, i in annotation.value
+                  mark._key ?= Math.random()
+
+                  if skippedMarks < @props.classification._hideMarksBefore and not @props.classification.completed
+                    skippedMarks += 1
+                    continue
+
+                  if mark._copy?
+                    continue
+
+                  toolDescription = taskDescription.tools[mark.tool]
+
+                  toolEnv =
+                    containerRect: @props.containerRect
+                    scale: @props.scale
+                    disabled: isPriorAnnotation
+                    selected: mark is @state.selection and not isPriorAnnotation
+                    getEventOffset: @props.getEventOffset
+                    preferences: @props.preferences
+
+                  toolProps =
+                    classification: @props.classification
+                    mark: mark
+                    details: toolDescription.details
+                    color: toolDescription.color
+                    size: toolDescription.size
+
+                  toolMethods =
+                    onChange: @handleChange.bind this, i
+                    onSelect: @handleSelect.bind this, annotation, mark
+                    onDeselect: @handleDeselect
+                    onDestroy: @handleDestroy.bind this, annotation, mark
+
+                  ToolComponent = drawingTools[toolDescription.type]
+                  <ToolComponent key={mark._key} {...toolProps} {...toolEnv} {...toolMethods} />}
+            </g>
+          else   
+            <g key={annotation._key} className="marks-for-annotation" data-disabled={isPriorAnnotation || null}>
+                {for mark, i in annotation.value when parseInt(mark.frame) is parseInt(@props.frame)
```

Comment:
If you change this to `when @props.workflow.configuration.multi_image_tool or parseInt(mark.frame) is parseInt(@props.frame)` then I think you can get rid of the outer `if` block and remove a lot of duplicated code here.